pkg/jks/jks.go

```diff
@@ -0,0 +1,160 @@
+/*
+Package jks provides routines for manipulating Java Keystore files.
+*/
+package jks
+
+import (
+	"crypto/sha1"
+	"crypto/x509"
+	"time"
+	"unicode/utf16"
+)
+
+const (
+	// MagicNumber is written at the start of each .jks file.
+	MagicNumber uint32 = 0xFEEDFEED
+
+	// DigestSeparator is used to build the file's verification digest. The
+	// digest is over the keystore password encoded as UTF-16, then this
+	// string (yes, really — check the OpenJDK source) encoded as UTF-8, and
+	// then the actual file data.
+	DigestSeparator = "Mighty Aphrodite"
+
+	// CertType is the certificate type string that is encoded into each
+	// certificate's header in the keystore.
+	CertType = "X.509"
+)
+
+// Keystore represents a single JKS file. It holds a list of certificates and a
+// list of keypairs (private keys with associated certificate chains).
+type Keystore struct {
+	// Certs is a list of CA certificates to trust. It may contain either
+	// root or intermediate CA certificates. It should not contain end-user
+	// certificates.
+	Certs []*Cert
+
+	// Keypairs is a list of private keys. Each key may have a certificate
+	// chain associated with it.
+	Keypairs []*Keypair
+}
+
+// Options for manipulating a keystore. These allow the caller to specify the
+// password(s) used, or to skip the digest verification if the password is
+// unknown.
+type Options struct {
+	// Password is used as part of a SHA-1 digest over the .jks file.
+	Password string
```

Comment:
we should strip the part that deals with password. We at least in foreseeable future wont take password as input to open the jks so its probably better to have the code at all.